Neos.Flow/Classes/ResourceManagement/PersistentResource.php

```diff
@@ -125,6 +133,16 @@ class PersistentResource implements ResourceMetaDataInterface, CacheAwareInterfa
      */
     protected $temporaryLocalCopyPathAndFilename;
 
+    /**
+     * The source is required for deferred persistence (after validation) in the framework and should otherwise not be set.
+     *
+     * @param string|array|resource|null $source
+     */
+    public function __construct($source = null)
```

Comment:
ooof, especially with the multi type I am not a huge fan of this. ESPECIALLY as this thing is called persisted resource and it is not persisted at all if this is used with the temporary files. In fact someone could come up with the silly idea of actually creating a PersistedResource this way and just handing it to the repository (which obviously fails miserably). I don't like this change and would rather introduce something else for this use case...

Reply:
I see your point here. What about a static constructor method that creates such a deferred persistent resource? Generally, you should not manually create such a deferred resource (you probably should never create a `PersistentResource` manually at all).